runtime/lua/vim/lsp/handlers.lua

```diff
@@ -611,6 +615,24 @@ M['window/showDocument'] = function(_, result, ctx, _)
   return { success = success or false }
 end
 
+---@see https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_inlayHint_refresh
+M['workspace/inlayHint/refresh'] = function(err, _, ctx)
+  if err then
+    return vim.NIL
+  end
+
+  for _, bufnr in ipairs(vim.lsp.get_buffers_by_client_id(ctx.client_id)) do
+    for _, winid in ipairs(api.nvim_list_wins()) do
+      if api.nvim_win_get_buf(winid) == bufnr then
+        require('vim.lsp.inlay_hint').refresh({ bufnr = bufnr })
+        break
+      end
+    end
+  end
```

Comment:
If a server sends a refresh, this looks like it'll add inlay hints to visible buffers even though this feature is opt in, and could be surprising.

I think this handler should invalidate the cache by setting version to nil for all buffers for this client. Then, for visible buffers that already had cached results, call refresh. Otherwise, clear the namespace.

This would require users to have something in place that could fire off a new request if they make a buffer that they had inlay hints enabled for visible again after a refresh, but because this is opt in, I think that's fine.